routing/router.go

```diff
@@ -564,14 +566,43 @@ func (r *ChannelRouter) processUpdate(msg interface{}) error {
 		chanUtxo, err := r.cfg.Chain.GetUtxo(&fundingPoint.Hash, fundingPoint.Index)
 		if err != nil {
 			return errors.Errorf("unable to fetch utxo for "+
-				"chan_id=%v: %v", channelID, err)
+				"chan_id=%v: %v", msg.ChannelID, err)
+		}
+
+		// Recreate witness output to be sure that declared in
+		// channel edge bitcoin keys and channel value corresponds to
+		// the reality.
+		_, witnessOutput, err := lnwallet.GenFundingPkScript(
+			msg.BitcoinKey1.SerializeCompressed(),
+			msg.BitcoinKey2.SerializeCompressed(),
+			chanUtxo.Value,
+		)
+		if err != nil {
+			return errors.Errorf("unable to create funding pk "+
+				"script: %v", err)
+		}
+
+		// By checking the equality of witness pkscripts we checks that
+		// funding witness script is multisignature lock which contains
+		// both local and remote public keys which was declared in
+		// channel edge and also that the announced channel value is
+		// right.
+		class := txscript.GetScriptClass(chanUtxo.PkScript)
```

Comment:
The call to `GetScriptClass` is unnecessary, instead the scripts can simply be compared directly.